components/server/src/session-handler.ts

```diff
@@ -43,11 +43,25 @@ export class SessionHandlerProvider {
     }
 
     protected getCookieOptions(env: Env): express.CookieOptions {
-        const hostParts = env.hostUrl.url.host.split('.');
-        const baseDomain = hostParts.slice(hostParts.length - 2).join('.');
-        let domain = `.${baseDomain}`;
+        const hostName = env.hostUrl.url.host;
+
+        let domain = hostName;
+        if (env.devBranch) {
+            // Use cookie for base domain to allow cookies being sent via ingress proxy in preview environments
+            //
+            // Otherwise, clients (in this case Chrome) may ignore (as in: save it, but don't send it on consequent requests) the 'Set-Cookie:...' send with a redirect (302, to github oauth)
+            // For details, see:
+            // - RFC draft sameSite: http://httpwg.org/http-extensions/draft-ietf-httpbis-cookie-same-site.html
+            // - https://bugs.chromium.org/p/chromium/issues/detail?id=150066
+            // - google: chromium not sending cookies set with redirect
+            
+            const hostParts = hostName.split('.');
+            const baseDomain = hostParts.slice(hostParts.length - 2).join('.');
+            domain = `.${baseDomain}`;
```

Comment:
I think we need the dot prefixing in all cases, as we want tosend the cookie with requests to subdomains as well, no?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

cf. Domain

> Contrary to earlier specifications, leading dots in domain names (.example.com) are ignored.

Reply:
then lets remove the dot here as well.